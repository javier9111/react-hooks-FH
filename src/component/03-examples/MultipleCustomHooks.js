import React from "react";
import { useFetch } from "../../Hooks/useFetch";
import "../02-useEffect/effects.css";
export const MultipleCustomHooks = () => {
  const { data, loading, error } = useFetch(
    `https://www.breakingbadapi.com/api/quotes/1`
  );
  // console.log(state);
  const { author, quote } = !!data && data[0];
  return (
    <div>
      <h1>Breaking bad Quotes</h1>
      <hr />
      {loading ? (
        <div className="alert alert-info text-center">loading...</div>
      ) : (
        <blockquote className="blockquote text-right">
          <p className="mb-0">{quote}</p>
          <footer className="blockquote-footer">{author}</footer>
        </blockquote>
      )}
    </div>
  );
};
