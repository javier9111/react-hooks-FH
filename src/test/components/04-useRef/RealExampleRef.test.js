import React from "react";
import { shallow } from "enzyme";
import { RealExampleRef } from "../../../component/04-useRef/RealExampleRef";
import { MultipleCustomHooks } from "../../../component/03-examples/MultipleCustomHooks";

describe("Test on <RealExampleRef/>", () => {
  test("should show correctly", () => {
    const wrapper = shallow(<RealExampleRef />);

    expect(wrapper).toMatchSnapshot();
  });

  test("should show the component <MultipleCustomHooks/>", () => {
    const wrapper = shallow(<RealExampleRef />);
    const button = wrapper.find("button");

    //console.log(wrapper.html());
    button.simulate("click");
    //expect(wrapper.contains(<MultipleCustomHooks />)).toBe(true); this an alternative to the assertion.
    expect(wrapper.find(MultipleCustomHooks).exists()).toBe(true);
  });
});
