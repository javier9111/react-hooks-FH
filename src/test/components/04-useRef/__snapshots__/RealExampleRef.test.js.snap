// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <RealExampleRef/> should show correctly 1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
